src/TextField/TextFieldLabel.js

```diff
@@ -58,32 +63,34 @@ const TextFieldLabel = (props) => {
     shrink,
     htmlFor,
     style,
+    focusStyle,
     onTouchTap,
   } = props;
 
-  const styles = {
-    root: {
-      position: 'absolute',
-      lineHeight: '22px',
-      top: 38,
-      transition: transitions.easeOut(),
-      zIndex: 1, // Needed to display label above Chrome's autocomplete field background
-      cursor: disabled ? 'default' : 'text',
-      transform: shrink ?
-        'perspective(1px) scale(0.75) translate3d(0, -28px, 0)' :
-        'scale(1) translate3d(0, 0, 0)',
-      transformOrigin: 'left top',
-      pointerEvents: shrink ? 'none' : 'auto',
-      userSelect: 'none',
-    },
-  };
-
   const {prepareStyles} = muiTheme;
 
+  const focusStyles = shrink ? Object.assign({}, {
+    transform: 'perspective(1px) scale(0.75) translate3d(0, -28px, 0)',
+    pointerEvents: 'none',
+  }, focusStyle) : null;
+
+  const styles = Object.assign({}, {
```

Comment:
Could we follow [this pattern](https://github.com/callemall/material-ui/blob/master/src/Snackbar/Snackbar.js#L7) for writing style? We plan to improve how we style the component. Be consistent will help us.